### ISSUE TITLE
Dependency solving fails for dependencies of git dependencies

### ISSUE THREAD

**Issue by rphes** [sdispater/poetry#788]
The title is somewhat cryptic, but I will explain it below.

- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS Mojave 10.14.2
- **Poetry version**: 0.12.10
- **Python version**: 3.7.0

## Issue
Suppose we have two repositories: _A_ and _B_. Project _A_ lives in a git repository and _B_ depends on _A_ by means of a git dependency: _B_ --git--> _A_. If we add a new dependency _C_ to project _A_, push the updated `pyproject.toml` and `poetry.lock` to our remote and then use `poetry install` on project _B_, we get a `SolverProblemError` saying that dependency _C_ cannot be resolved.
I think we'd expect poetry to simply install dependency _C_ and successfully finish executing.

## Steps to reproduce
- Create the two projects:
```bash
poetry new project-a && poetry new project-b
```
and make sure `project-a` is pushed to an accessible git repository
- Add a dependency for `project-a` to `project-b` by adding this line to its `pyproject.toml`:
```toml
project-a = { git = "git@<git repo>/project-a.git" }
```
and then
```bash
poetry install
```
- Now add a dependency to `project-a`, e.g.
```bash
poetry add pendulum
```
and push the changes to the repository
- Finally, in `project-b`, again use 
```bash
poetry install
```
and you should see something like this:

```
[SolverProblemError]
Because no versions of project-a match !=0.1.0
 and project-a (0.1.0) depends on pendulum (^2.0), every version of project-a requires pendulum
 (^2.0).
So, because no versions of pendulum match >=2.0,<3.0
 and project-b depends on project-a (*), version solving failed.
```

## Workarounds
1. Also add `pendulum` to `project-b`'s dependencies
2. Use `poetry update` in `project-b` first. I am not sure why this works, but it does:
```bash
...
  - Installing pendulum (2.0.4)
  - Updating project-a (0.0.0 c0fe7af -> 0.1.0 dcfe7ff)
```
Also note the version number on `project-a`, which should be 0.1.0, also before updating.

I'm glad to provide any additional information.

**Comment by sdispater** [comment]
This is the intended behavior. Basically `poetry install` uses only the lock file if it exists and in this case is not aware of any package named `pendulum` since it is missing from the lock file.

That's the reason why `poetry update` works since in this case Poetry will update the lock file and add `pendulum` to it.

**Comment by rphes** [comment]
Alright, that makes sense. I guess I got confused by the fact that the second `poetry install` causes the git dependency to be updated, after which the `pendulum` dependency breaks. This is being addressed in #691, right?
